### PR TITLE
Add test function

### LINKED ISSUE #436: Add pydatastructs.test function to run tests
#### Description of the problem

The above function would make it easier for the user to verify the installation of PyDataStructs APIs.

### PR DESCRIPTION
**Pull request by sak-codes**
#### References to other Issues or PRs or Relevant literature

Fixes #436

### CODE REVIEW

**Review by czgdp1807 — COMMENTED** [review]
What is this for?

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Use it inside this PR.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Upgrade it to 15 from 10.

https://github.com/codezonediitj/pydatastructs/blob/bab603f0b4f7571ef9a96f170412231b1a3d4752/.github/workflows/ci.yml#L145

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Following features should be added,

1. Benchmark tests should be run only when needed. `benchmark=False` should turn them off.
2. Benchmark tests if run should compulsorily accept an argument for benchmark size.
3.
```
>>> from pydatastructs import linear_data_structures as lds
>>> test(submodules=lds)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/czgdp1807/codezonediitj/pydatastructs/pydatastructs/utils/test_util.py", line 43, in test
    for sub in submodules:
TypeError: 'module' object is not iterable
```

**Comment by sak-codes** [comment]
I have tried to complete all the 3 features. Please see the output of last point:
```
>>> import pydatastructs as pyds
>>> from pydatastructs import linear_data_structures as lds
>>> pyds.test(lds)
============================= test session starts ==============================
platform darwin -- Python 3.8.16, pytest-7.3.1, pluggy-1.0.0
rootdir: /Users/thebigbool/repos/pydatastructs
plugins: cov-4.1.0, xdist-3.3.1, anyio-3.7.0
collected 31 items                                                             

pydatastructs/linear_data_structures/tests/test_arrays.py ...            [  9%]
pydatastructs/linear_data_structures/tests/test_linked_lists.py .....    [ 25%]
pydatastructs/linear_data_structures/tests/test_algorithms.py .......... [ 58%]
.............                                                            [100%]

============================== 31 passed in 0.10s ==============================
```

**Review by Smit-create — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
 
       - name: Run tests
         run: |
-          python -m pytest --doctest-modules --cov=./ --cov-report=xml -s
```
[the remark this answers is not in the document]
Comment (sak-codes) [reply .github/workflows/ci.yml:96]: Coverage is handled already on line 46-56. So I didn't change that command.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**